### PR TITLE
remove log message that causes deadlock

### LINKED ISSUE #94: deadlock in WriteEntryTimeout
IngestMuxer.WriteEntryTimeout can be called via NegotiateTag, which already holds the IngestMuxer.mtx for writing. WriteEntryTimeout attempts to get a read lock on mtx, causing a deadlock.

muxer.go:770
muxer.go:491

### PR DESCRIPTION
**Pull request by djfritz**
Fixes #94

### CODE REVIEW

File: ingest/muxer.go [hunk 1]
```diff
@@ -536,7 +536,6 @@ func (im *IngestMuxer) NegotiateTag(name string) (tg entry.EntryTag, err error)
 			remoteTag, err := v.NegotiateTag(name)
 			if err != nil {
 				// something went wrong, kill it and let it re-initialize
-				im.Error("NegotiateTag on %v for %v: %v", v.conn.RemoteAddr(), name, err)
```
Comment (traetox) [inline ingest/muxer.go:539]: this log might be super handy, should we:
```
mtx.Unlock()
im.Error("NegotiateTag on %v for %v: %v", v.conn.RemoteAddr(), name, err)
mtx.Lock()
```

?  If that has other affects i am not seeing, feel free to reject

**Review by traetox — COMMENTED** [review]
LGTM

**Comment by djfritz** [comment]
@traetox PTAL

**Review by traetox — APPROVED** [review]
LGTM

**Pull request merged**